### PR TITLE
feat(karma-esm): enable testing with modules on all browsers

### PR DESCRIPTION
**Pull request by LarsDenBakker**
Relies on https://github.com/open-wc/open-wc/pull/553 and https://github.com/open-wc/open-wc/pull/563

- Update `karma-esm` to enable testing with modules on all browsers, including IE11. 
- Update testing-karma to use new version of karma-esm
- Remove legacy testing-karma config with karma-webpack
- Add import map option, which shims/polyfills where necessary. we need to discuss the workflow here though, since we can't read it from an index.html

`karma-esm` now spins up `es-dev-server` to handle code compilation, and uses `@open-wc/building-utils/index-html` to set up polyfills etc. just like the rest of our tools.

Unfortunately it was a bit more work than I'd hoped, we have to set up a custom karma test runner html and run a somewhat elaborate loader script to load all the polyfills, libs and run the tests. This is perhaps something to improve on later, but it works and it's not something users notice.

To see it in action, look at `/packages/karma-esm/demo/` and run the test commands from `karma-esm`.

Closes https://github.com/open-wc/open-wc/issues/485
Fixes https://github.com/open-wc/open-wc/issues/484, https://github.com/open-wc/open-wc/issues/481

TODO:

- [x] add ability to exclude some files from babel compile, large libs like mocha, sinon and chai are slow to compile and don't need to be processed for modern browsers
- [x] update documentation
- [x] watch served files
- [x] test on windows

### CODE REVIEW

**Comment by LarsDenBakker** [comment]
Should we consider this a breaking change? The API didn't change, but a lot changed under the hood.

It does break for people using a custom webpack configuration, should we consider that part of the public API?

File: packages/testing-karma/default-config.js [hunk 1]
```diff
@@ -1,6 +1,115 @@
...
+
+    // fallback: resolve any karma- plugins
+    'karma-*',
+  ],
+
+  frameworks: ['esm', 'mocha', 'snapshot', 'mocha-snapshot', 'source-map-support'],
+
+  esm: {
+    coverage,
+    compatibility: compatibility ? 'all' : 'none',
```
Comment (daKmoR) [inline packages/testing-karma/default-config.js]: why only support "all" or "none" while the es-dev-server supports more?

**Comment by daKmoR** [comment]
woohooo - that really looks nice 🎉 all puzzle pieces are falling into place 🤗 

imho it should definitely be a breaking change... especially as we are deleting 2 files which have been probably used as entry points `esm-config.js`, `legacy-config.js`.

and I would say we should also take that chance to export all configs via a `index.js` e.g. end-user api should be
```js
// old
const createDefaultConfig = require('@open-wc/testing-karma/default-config.js');

// new
const createDefaultConfig = require('@open-wc/testing-karma');
```

and move "specific" configs into `src` so it's more clear what people should use as entry points

File: packages/testing-karma/default-config.js (hunk 1 above)
Comment (LarsDenBakker) [reply packages/testing-karma/default-config.js]: I was thinking that for tests the speed is a bit less important than for developing. The primary speed difference between compatibility modes is whether or not babel has to be used, which is very quickly the case for code coverage.

But we could also just make it a flag like es-dev-server, that does make it consistent.

Remember that this is just the default config, `karma-esm` supports all the flags and people could turn it on in their config.

**Comment by LarsDenBakker** [comment]
True you could have imported the separate configs. Let's make it a non-default export then:
```
const { createDefaultConfig } = require('@open-wc/testing-karma');
```

that way we have a road forward for other types of configs later on.

**Comment by LarsDenBakker** [comment]
Added another TODO: test on windows

File: packages/testing-karma/default-config.js (hunk 1 above)
Comment (LarsDenBakker) [reply packages/testing-karma/default-config.js]: Made it an enum flag instead.

**Comment by LarsDenBakker** [comment]
Tested everything on windows and resolved latest change requests/TODOs. 

Split this change up into two separate breaking commits @daKmoR please check if the commit messages look correct.

This is ready for merging now!

File: packages/karma-esm/TODO.md [hunk 2]
```diff
@@ -0,0 +1,2 @@
+- don't run babel on mocha etc. it's now passed to es dev server
```
Comment (daKmoR) [inline packages/karma-esm/TODO.md]: we probably want to remove this file at some point right?

Comment (LarsDenBakker) [reply packages/karma-esm/TODO.md]: Yeah that file is no longer needed! I've removed

**Comment by daKmoR** [comment]
version updates do not look that good 🙈 

```
 - @open-wc/building-rollup: 0.8.4 => 0.8.5
 - @open-wc/building-utils: 2.4.1 => 3.0.0
 - @open-wc/building-webpack: 2.3.1 => 2.3.2
 - @open-wc/chai-dom-equals: 0.12.20 => 1.0.0
 - es-dev-server: 1.6.3 => 2.0.0
 - @open-wc/karma-esm: 1.1.5 => 2.0.0
 - rollup-plugin-index-html: 1.4.2 => 2.0.0
 - @open-wc/semantic-dom-diff: 0.13.0 => 1.0.0
 - @open-wc/testing-helpers: 1.0.11 => 2.0.0
 - @open-wc/testing-karma-bs: 1.1.16 => 2.0.0
 - @open-wc/testing-karma: 2.0.16 => 3.0.0
 - @open-wc/testing: 1.0.13 => 2.0.0
 - @open-wc/webpack-index-html-plugin: 1.2.6 => 2.0.0
```


ok I now made separate commits for the breaking changes and the updates so we get the version updates we need 🤗 

```
 - @open-wc/building-rollup: 0.8.4 => 0.8.5
 - @open-wc/building-utils: 2.4.1 => 2.4.2
 - @open-wc/building-webpack: 2.3.1 => 2.3.2
 - @open-wc/chai-dom-equals: 0.12.20 => 0.12.21
 - es-dev-server: 1.6.3 => 1.6.4
 - @open-wc/karma-esm: 1.1.5 => 2.0.0
 - rollup-plugin-index-html: 1.4.2 => 1.4.3
 - @open-wc/semantic-dom-diff: 0.13.0 => 0.13.1
 - @open-wc/testing-helpers: 1.0.11 => 1.0.12
 - @open-wc/testing-karma-bs: 1.1.16 => 1.1.17
 - @open-wc/testing-karma: 2.0.16 => 3.0.0
 - @open-wc/testing: 1.0.13 => 1.0.14
 - @open-wc/webpack-index-html-plugin: 1.2.6 => 1.2.7
```

**Pull request merged**